[ARM64][Windows 11]: Running Freshly Created Avalonia 11.0.0-preview4/5 application returns a transparent window

**Describe the bug**
Running Avalonia applications using 11.0.0-preview4,preview-5 and even nightly builds on **native** Windows on Arm devices (Surface Pro X / Windows Dev Kit 2023) fail to properly display window contents. By default, only the outer frame of the window is created while the inner contents are completely transparent.

**To Reproduce**
Steps to reproduce the behavior:

1. Run `dotnet new avalonia.app -f net7.0 -av 11.0.0-preview4` from an Arm64 Windows 11 device
2. Run `dotnet publish -c Release -r win10-arm64`
3. Execute the exe from the published output directory
4. Notice the app runs with a transparent window 
5. Update the dependencies to 11.0.0-preview5
6. Remove Mode property from <FluentTheme> declaration in App.axml
7. Execute the same process
8. Notice the transparent window exists


**Expected behavior**
Creating a new application from the Avalonia.Templates should properly run without modification for Windows Arm64.

**Screenshots**
If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/4016293/220016037-76d311a5-d62a-44e1-9ef4-bf26476a28ec.png)

**Desktop (please complete the following information):**

- OS: Windows 11 22530 Arm64
- Version 11.0.0-preview5

**Additional context**
The interesting thing is deleting `av_libglesv2.dll` from the output directory will cause the app to load properly. I'm assuming this is a composition issue related to  #8776 ?

![image](https://user-images.githubusercontent.com/4016293/220017292-6cec0988-9d71-4986-af48-f69f6ad9bc81.png)

I'm not sure why I'd look for a 11.1.999 version when the PR build says to use a 11.0.999 version. But I tried changing the 0 to 1 as you seemed to suggest. It still fails:

>  error NU1102: Unable to find package Avalonia.Diagnostics with version (>= 11.1.999-cibuild0041739-beta)
> error NU1102:   - Found 303 version(s) in avalonia-nightly [ Nearest version: 11.1.999-cibuild0041722-beta ]

--- a/src/Windows/Avalonia.Win32/OpenGl/Angle/AngleWin32EglDisplay.cs
+++ b/src/Windows/Avalonia.Win32/OpenGl/Angle/AngleWin32EglDisplay.cs
@@ -85,10 +85,7 @@ namespace Avalonia.Win32.OpenGl.Angle
                     chosenAdapter = adapters
                         .OrderByDescending(x =>
                             // Put adreno in lower priority - it's broken in Avalonia.
-                            x.name.Contains("adreno") ? -1
-                            : x.name.Contains("nvidia") ? 2
-                            : x.name.Contains("amd") ? 1
-                            : 0)
+                            x.name.Contains("adreno") ? -1 : 0)
                         .First().adapter
                         .CloneReference();
 

--- a/src/Windows/Avalonia.Win32/OpenGl/Angle/AngleWin32EglDisplay.cs
+++ b/src/Windows/Avalonia.Win32/OpenGl/Angle/AngleWin32EglDisplay.cs
@@ -46,8 +46,7 @@ namespace Avalonia.Win32.OpenGl.Angle
             }, AngleOptions.PlatformApi.DirectX11);
         }
 
-        public static unsafe AngleWin32EglDisplay CreateD3D11Display(Win32AngleEglInterface egl,
-            bool preferDiscreteAdapter = false)
+        public static unsafe AngleWin32EglDisplay CreateD3D11Display(Win32AngleEglInterface egl)
         {
             var featureLevels = new[]
             {
@@ -65,7 +64,9 @@ namespace Avalonia.Win32.OpenGl.Angle
                 using var factory = MicroComRuntime.CreateProxyFor<IDXGIFactory1>(pDxgiFactory, true);
 
                 void* pAdapter = null;
-                if (preferDiscreteAdapter)
+                // As for now, we only need to redefine default adapter only on ARM64 just in case of Adreno GPU.
+                var redefineDefaultAdapter = RuntimeInformation.ProcessArchitecture == Architecture.Arm64;
+                if (redefineDefaultAdapter)
                 {
                     ushort adapterIndex = 0;
                     var adapters = new List<(IDXGIAdapter1 adapter, string name)>();
@@ -80,9 +81,17 @@ namespace Avalonia.Win32.OpenGl.Angle
 
                     if (adapters.Count == 0)
                         throw new OpenGlException("No adapters found");
+
                     chosenAdapter = adapters
-                        .OrderByDescending(x => x.name.Contains("nvidia") ? 2 : x.name.Contains("amd") ? 1 : 0)
-                        .First().adapter.CloneReference();
+                        .OrderByDescending(x =>
+                            // Put adreno in lower priority - it's broken in Avalonia.
+                            x.name.Contains("adreno") ? -1
+                            : x.name.Contains("nvidia") ? 2
+                            : x.name.Contains("amd") ? 1
+                            : 0)
+                        .First().adapter
+                        .CloneReference();
+
                     foreach (var a in adapters)
                         a.adapter.Dispose();
                 }

